pkg/mcs/exportspec_test.go

```diff
@@ -136,25 +137,83 @@ func TestCompatibility(t *testing.T) {
 		other      *mcs.ExportSpec
 		compatible bool
 		field      string
+		mutate     func(mcs.ExportSpec) *mcs.ExportSpec
 	}{
-		"compatible: empty": {spec: &mcs.ExportSpec{}, other: &mcs.ExportSpec{}, compatible: true},
-		/*
-			"compatible: identical":        {},
-			"compatible: local properties": {},
-			"conflicting: type":            {},
-			"conflicting: affinity":        {},
-			"conflicting: port":            {},
-		*/
+		"compatible: empty": {spec: &mcs.ExportSpec{}, compatible: true,
+			mutate: func(es mcs.ExportSpec) *mcs.ExportSpec {
+				return &mcs.ExportSpec{}
+			}},
+
+		"compatible: identical": {spec: &sample, compatible: true,
+			mutate: func(es mcs.ExportSpec) *mcs.ExportSpec {
+				return &es
+			}},
+
+		"compatible: cluserID": {spec: &sample, compatible: true,
+			mutate: func(es mcs.ExportSpec) *mcs.ExportSpec {
+				es.ClusterID = "different_cluster"
+				return &es
+			}},
+		"compatible: CreatedAt": {spec: &sample, compatible: true,
+			mutate: func(es mcs.ExportSpec) *mcs.ExportSpec {
+				es.CreatedAt.Time = es.CreatedAt.Time.Add(1 * time.Second)
+				return &es
+			}},
+		"conflicting: name": {spec: &sample, compatible: false, field: "name",
+			mutate: func(es mcs.ExportSpec) *mcs.ExportSpec {
+				es.Name = "different_name"
+				return &es
+			}},
+		"conflicting: namespace": {spec: &sample, compatible: false, field: "namespace",
+			mutate: func(es mcs.ExportSpec) *mcs.ExportSpec {
+				es.Namespace = "different_namepsace"
+				return &es
+			}},
+
+		"conflicting: type": {spec: &sample, compatible: false, field: "type",
+			mutate: func(es mcs.ExportSpec) *mcs.ExportSpec {
+				es.Service.Type = mcsv1a1.Headless
+				return &es
+			}},
+
+		"conflicting: affinity": {spec: &sample, compatible: false, field: "affinity",
+			mutate: func(es mcs.ExportSpec) *mcs.ExportSpec {
+				es.Service.SessionAffinity = corev1.ServiceAffinityNone
+				return &es
+			}},
+
+		"conflicting: affinity config": {spec: &sample, compatible: false, field: "affinity config",
+			mutate: func(es mcs.ExportSpec) *mcs.ExportSpec {
+				es.Service.SessionAffinityConfig = &corev1.SessionAffinityConfig{
+					ClientIP: &corev1.ClientIPConfig{
+						TimeoutSeconds: &otherTimeout,
```

Comment:
no need to make `otherTimeout` global. It should be a local variable in this (unnamed) function